lib/engine/step/g_1846/train.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require_relative '../train'
+
+module Engine
+  module Step
+    module G1846
+      class Train < Train
+        def setup
+          super
+          @issued = false
+        end
+
+        def actions(entity)
+          return [] if entity.minor? || entity.company?
+
+          return [] if entity.owner == @game.share_pool
+
+          if entity.corporation? && must_buy_train?(entity)
+            actions_ = %w[buy_train]
+            actions_ += %w[buy_shares sell_shares] if can_issue?(entity)
+            return actions_
+          elsif entity == current_entity.owner
+            return %w[sell_shares] unless can_issue?(current_entity)
+          end
+
+          return %w[buy_train pass] if can_buy_train?(entity)
+
+          []
+        end
+
+        def skip!
+          @round.attempt_to_buy_train_no_president(self, :process_buy_train)
+        end
+
+        # ugly hack since the view for IssueShares keys off the buy_shares
+        # action being included
+        def redeemable_shares(_entity)
```

Comment:
hm? can you change the view? or is this fine

Reply:
I changed the view